kura/org.eclipse.kura.core/src/main/java/org/eclipse/kura/core/db/H2DbServer.java

```diff
@@ -0,0 +1,126 @@
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.core.db;
+
+import java.sql.SQLException;
+import java.util.Map;
+
+import org.eclipse.kura.configuration.ConfigurableComponent;
+import org.h2.tools.Server;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class H2DbServer implements ConfigurableComponent {
+
+    private enum ServerType {
+        WEB,
+        TCP,
+        PG
+    };
+
+    private static final Logger logger = LoggerFactory.getLogger(H2DbServer.class);
+    private Server server;
+
+    protected void activate(Map<String, Object> properties) {
+        logger.info("activating...");
+        updated(properties);
+        logger.info("activating...done");
+    }
+
+    protected void updated(Map<String, Object> properties) {
+        logger.info("updating...");
+        restartServer(new ServerConfiguration(properties));
+        logger.info("updating...done");
+    }
+
+    protected void deactivate() {
+        logger.info("deactivating...");
+        shutdownServer();
+        logger.info("deactivating...done");
+    }
+
+    private void restartServer(ServerConfiguration configuration) {
+        shutdownServer();
+        Server server = null;
+        if (configuration.isServerEnabled) {
+            try {
+                logger.info("Starting DB server...");
+                final String[] commandline = configuration.serverCommandLine.split(" ");
+                logger.debug("Server type: {}, commandline: {}", configuration.serverType, commandline);
+                switch (configuration.serverType) {
+                case TCP:
+                    server = Server.createTcpServer(commandline);
+                    break;
+                case WEB:
+                    server = Server.createWebServer(commandline);
+                    break;
+                case PG:
+                    server = Server.createPgServer(commandline);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Unknown server type");
+                }
+                server.start();
+                this.server = server;
+                logger.info("Starting DB server...done");
+            } catch (SQLException e) {
+                logger.error("Failed to start server", e);
+                shutdownServer(server);
+            }
+        }
+    }
+
+    private void shutdownServer() {
+        shutdownServer(this.server);
+        this.server = null;
+    }
+
+    private void shutdownServer(Server server) {
+        if (server != null) {
+            try {
+                logger.info("Shutting down DB server...");
+                server.stop();
+                logger.info("Shutting down DB server...done");
+            } catch (Exception e) {
+                logger.error("failed to shutdown DB server", e);
+            }
+        }
+    }
+
+    private static class ServerConfiguration {
```

Comment:
I believe we are standardising on using a public "Options" class for the configuration.
I also believe it would be safer to start using defensive code:
1. Inizialize with safe defaults if the property is missing or unexpected instanceof
2. Check instanceof before casting